### PR TITLE
[MBL1260] Add Checkout Screen to Flow

### PR DESCRIPTION
**Pull request by mtgriego**
# 📲 What

Adding the checkout screen to the late pledge flow.  Currently will run through the calls needed to make a purchase, but does not proceed to thanks page afterwards

# 🤔 Why

Its the last step and screen in the flow

# 🛠 How

See code

# 👀 See

Screenshot_20240327-101728

# 📋 QA

Go through a late pledge flow and reach this screen.  To monitor calls use logcat

# Story 📖

[MBL-1260](https://kickstarter.atlassian.net/browse/MBL-1260)

[MBL-1260]: https://kickstarter.atlassian.net/browse/MBL-1260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                    apolloClient.getStoredCards().asFlow().map { cards ->
+                        storedCards = cards
+                        mutableLatePledgeCheckoutUIState.emit(
+                            LatePledgeCheckoutUIState(
+                                storeCards = storedCards,
+                                userEmail = userEmail,
+                            )
+                        )
+                    }.catch {
+                        // Some error
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt:75]: do we have an error path for stored cards not fetching?

Comment (mtgriego) [reply app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt:75]: not really, i think we would just throw out a generic error, but ill double check to see what the current flow does

Comment (Arkariang) [reply app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt:75]: on PledgeFragment we do the generic error, red snackbar with "something went wrong". Same pattern could be use throughout the late pledge chedkout flow. And can be a separated ticket :)

Comment (leighdouglas) [reply app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt:75]: I'm cool with it being a separate ticket, just something that im currently trying to address in the QA script and was curious how we were handling it

**Review by Arkariang — DISMISSED** [review]

**Review by Arkariang — APPROVED** [review]

File: app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt (hunk 1 above)
Comment (Arkariang) [reply app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt:75]: [MBL-1306](https://kickstarter.atlassian.net/browse/MBL-1306) ticket here

**Pull request merged**